### PR TITLE
Lbe 75 program

### PR DESCRIPTION
**Pull request by akshatakanaje**

### CODE REVIEW

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 1]
```diff
@@ -33,19 +36,20 @@ public class ClassController {
...
 	
 	//GetAllClasses
 	@GetMapping(value = "/allClasses", produces = "application/json")
 	@ApiOperation("Get all Class List")
-	private ResponseEntity<?> getAllClassesList()  throws ResourceNotFoundException
-	{ 
-		System.out.println("in getall classes");
+	public ResponseEntity<?> getAllClassesList()  throws ResourceNotFoundException
+	{
+		log.info("in get all classes");
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/controller/ClassController.java:52]: pls remove this log message

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 2]
```diff
@@ -54,41 +58,41 @@ private ResponseEntity<?> getAllClassesList()  throws ResourceNotFoundException
 	@GetMapping(path="class/{classId}", produces = "application/json")  
 	@ResponseBody
 	@ApiOperation("Get Class details by ID")
-	private ResponseEntity <?> getClassesById(@PathVariable("classId") @NotBlank @Positive Long classId)throws ResourceNotFoundException
-	{  
-		System.out.println("in get All classes by ClassId");
+	public ResponseEntity <?> getClassesById(@PathVariable("classId") @NotBlank @Positive Long classId)throws ResourceNotFoundException
+	{
+		log.info("in get All classes by ClassId");
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/controller/ClassController.java:63]: same, remove this too

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 3]
```diff
@@ -54,41 +58,41 @@ private ResponseEntity<?> getAllClassesList()  throws ResourceNotFoundException
...
-		{ 
-			System.out.println("in getall Class By ClassTopic");
-			List<ClassDto> ClassesList = classServices.getClassesByClassTopic(classTopic);
-			return ResponseEntity.ok(ClassesList);  
-		} 
+	@GetMapping(value = "/classes/{classTopic}", produces = "application/json")
+	@ApiOperation("Get all Classes by Class Topic ")
+	public ResponseEntity<?> getAllClassesByClassTopic(@PathVariable("classTopic") @NotBlank @Positive String classTopic)  throws ResourceNotFoundException
+	{
+		log.info("in getall Class By ClassTopic");
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/controller/ClassController.java:73]: same as above, remove this log entry

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 4]
```diff
@@ -101,19 +105,21 @@ private ResponseEntity <?> getClassesByStaffId(@PathVariable(value="staffId") @N
...
 	@ResponseBody
 	@ApiOperation("Delete Class by Class ID")
-	private ResponseEntity<?>  deleteByClassId(@PathVariable("classId")@NotBlank @Positive Long classId) throws ResourceNotFoundException  
-	{  
-	System.out.println("in delete by classId controller");
-	boolean deleted = classServices.deleteByClassId(classId); 
+	@PreAuthorize("hasRole('ADMIN')")
+	public ResponseEntity<?>  deleteByClassId(@PathVariable("classId")@NotBlank @Positive Long classId) throws ResourceNotFoundException
+	{
+		log.info("in delete by classId controller");
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/controller/ClassController.java:121]: remove the log entry

File: src/main/java/com/numpyninja/lms/security/WebSecurityConfig.java [hunk 5]
```diff
@@ -77,6 +76,9 @@ protected void configure(HttpSecurity http) throws Exception {
                 .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
                 .authorizeRequests().antMatchers("/login/**").permitAll()
                 .antMatchers(PUBLIC_URLS).permitAll()
+                .antMatchers("/admin/**").hasRole("ADMIN")
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/security/WebSecurityConfig.java:79]: I don't think we need these 3 statements, can you pls explain what is your reasoning for adding these ?

File: src/main/java/com/numpyninja/lms/services/ClassService.java [hunk 6]
```diff
@@ -27,8 +27,7 @@
 @Service
 public class ClassService {
 	@Autowired
-
-	private ClassRepository classRepository;
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/ClassService.java:31]: Can you explain why private is removed?

**Comment by akshatakanaje** [comment]
Lalitha is working on this

**Pull request closed without merging**